components/org.wso2.carbon.identity.oauth.uma.permission.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/permission/endpoint/PermissionApiServiceImpl.java

```diff
@@ -98,73 +109,131 @@ public Response requestPermission(ResourceModelDTO requestedPermission, MessageC
+     * List requested permissions.
+     *
+     * @param requestedPermission Start Index of the result set to enforce pagination
+     * @return List of requested resources
+     */
     private List<Resource> getPermissionTicketRequest(ResourceModelDTO requestedPermission) {
 
         List<Resource> resourceList = new ArrayList<>();
         requestedPermission.forEach(resourceModelInnerDTO -> {
             Resource resource = new Resource();
             resource.setResourceId(resourceModelInnerDTO.getResourceId());
             List<String> resourceScopesList = new ArrayList<>();
-            resourceModelInnerDTO.getResourceScopes().forEach(resourceScope -> {
-                resourceScopesList.add(resourceScope);
-            });
+            resourceScopesList.addAll(resourceModelInnerDTO.getResourceScopes());
             resource.setResourceScopes(resourceScopesList);
             resourceList.add(resource);
         });
         return resourceList;
     }
 
-    private void handleErrorResponse(Throwable throwable, boolean isServerException)
-            throws PermissionEndpointException {
+    /**
+     * Logs the error, builds a PermissionEndpointException with specified details and throws it.
+     *
+     * @param message error message
+     * @param umaException
+     * @throws PermissionEndpointException
+     */
+    private void handleErrorResponse(String message, UMAException umaException) throws PermissionEndpointException {
 
-        String code;
+        String code = umaException.getCode();
         String errorCode = null;
         Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
-        boolean isStatusOnly = true;
-        if (throwable instanceof UMAException) {
-            code = ((UMAException) throwable).getCode();
-        } else {
-            code = UMAConstants.ErrorMessages.ERROR_UNEXPECTED.getCode();
-        }
+        boolean isHTTPStatusOnly = true;
+        boolean isServerException = umaException instanceof UMAServerException;
         if (isServerException) {
-            if (throwable == null) {
-                log.error(status.getReasonPhrase());
-            } else {
-                log.error(status.getReasonPhrase(), throwable);
-            }
+            log.error(message, umaException);
         } else {
-            log.error("Client error while requesting permission ticket.", throwable);
-            if (code != null) {
-                if (HandleErrorResponseConstants.RESPONSE_DATA_MAP.containsKey(code)) {
-                    String statusCode = HandleErrorResponseConstants.RESPONSE_DATA_MAP.get(code)[0];
-                    errorCode = HandleErrorResponseConstants.RESPONSE_DATA_MAP.get(code)[1];
-                    status = Response.Status.fromStatusCode(Integer.parseInt(statusCode));
-                    isStatusOnly = false;
-                }
+            if (log.isDebugEnabled()) {
+                log.debug(message + " - " + umaException.getErrorLogMessage());
```

Comment:
since this is debug level, we can print 'e' also. 
If possible print an error or warn level log with "message + " - " + umaException.getErrorLogMessage()"